impl/sql/mutationstorage/queue.go

```diff
@@ -61,29 +61,44 @@ func (m *Mutations) Send(ctx context.Context, domainID string, update *pb.EntryU
 	return m.send(ctx, domainID, logID, mData, time.Now())
 }
 
-// randLog returns a random, enabled log for domainID.
-func (m *Mutations) randLog(ctx context.Context, domainID string) (int64, error) {
-	// TODO(gbelvin): Cache these results.
+// ListLogs returns a list of all logs for domainID, optionally filtered for writable logs.
+func (m *Mutations) ListLogs(ctx context.Context, domainID string, writable bool) ([]int64, error) {
+	var query string
+	if writable {
```

Comment:
Maybe do:
```
query := "SELECT ..."
if writable {
  query += " WHERE ..."
}
```

Reply:
It's good security practice to **not** dynamically modify query strings in order to avoid any possibility of SQL injection.  I believe some DB libraries can even enforce that the query string is a compile time constant.